ambari-web/app/controllers/wizard/step2_controller.js

```diff
@@ -197,22 +218,77 @@ App.WizardStep2Controller = App.WizardStepController.extend({
     return installedHostsName;
   }.property('content.hosts'),
 
+  /**
+  * get hosts that are installed manually
+  return
+  */
+  getManuallyInstalledHosts: function () {
+    App.ajax.send({
+      name: 'hosts.confirmed.install',
+      sender: this,
+      success: 'getManuallyInstalledHostsSuccessCallback'
+    });
+  }.observes('manualInstall'),
+
+  getManuallyInstalledHostsSuccessCallback(response) {
+    var installedHosts = [];
+    if (response.items.length > 0) {
+      response.items.forEach(function (item, indx) {
+        installedHosts.push({
+          'hostName': item.Hosts.host_name,
+          'cpuCount': item.Hosts.cpu_count,
+          'memory': (parseFloat(item.Hosts.total_mem) * 0.000001).toFixed(2) + " GB",
+          'freeSpace': (parseFloat(item.Hosts.disk_info[0].available) *  0.000001).toFixed(2) + " GB"
+        });
+      });
+      this.set('preRegisteredHostFound', Em.I18n.t('installer.step2.preRegistered.hostCount').format(installedHosts.length));
+      this.set('manuallyInstalledHosts', installedHosts);
+      this.set('noPreRegisteredHosts', false);
+    } else {
+      this.set('preRegisteredHostFound', Em.I18n.t('installer.step2.preRegistered.hostCount').format(0));
+      this.set('noPreRegisteredHosts', true);
+      this.set('manuallyInstalledHosts', []);
+    }
+  },
+
+  deleteRegisteredHost: function (event) {
+    var hostName = event.context;
+    App.ajax.send({
+      name: 'common.delete.registered.host',
+      sender: this,
+      data: {
+        hostName: event.context
+      },
+      success: 'deleteRegisteredHostSuccessCallback'
+    });
+  },
+
+  deleteRegisteredHostSuccessCallback: function (response) {
+    this.getManuallyInstalledHosts();
+  },
+
   /**
    * Set not installed hosts to the hostNameArr
    * @method updateHostNameArr
    */
   updateHostNameArr: function () {
-    this.set('hostNameArr', this.get('hostNames').trim().split(new RegExp("\\s+", "g")));
-    this.parseHostNamesAsPatternExpression();
-    this.get('inputtedAgainHostNames').clear();
-    var tempArr = [],
-        hostNameArr = this.get('hostNameArr');
-    for (var i = 0; i < hostNameArr.length; i++) {
-      if (!this.get('installedHostNames').contains(hostNameArr[i])) {
-        tempArr.push(hostNameArr[i]);
-      }
-      else {
-        this.get('inputtedAgainHostNames').push(hostNameArr[i]);
+    var tempArr = [];
+    if (this.get('manualInstall')) {
+      var hostNames = this.get('manuallyInstalledHosts').mapProperty('hostName');
+      tempArr.push(hostNames);
```

Comment:
You are pushing the whole array instead hostname string, should be:
tempArr = this.get('manuallyInstalledHosts').mapProperty('hostName');